content/docs/pipelines/caching.md

```diff
@@ -0,0 +1,110 @@
+With caching enabled, the system can skip a step that has already been executed which saves time and money. 
+
+## Disabling/enabling caching
+
+Cache is enabled by default after Kubeflow Pipelines 0.4. 
+These are instructions on disabling and enabling cache service:
+
+### Configure access to your Kubeflow cluster
+
+Use the following instructions to configure `kubectl` with access to your
+Kubeflow cluster. 
+
+1.  To check if you have `kubectl` installed, run the following command:
+
+    ```bash
+    which kubectl
+    ```
+
+    The response should be something like this:
+
+    ```bash
+    /usr/bin/kubectl
+    ```
+
+    If you do not have `kubectl` installed, follow the instructions in the
+    guide to [installing and setting up kubectl][kubectl-install].
+
+2.  Follow the [guide to configuring access to Kubernetes
+    clusters][kubectl-access]. 
+
+### Disabling caching in your Kubeflow Pipelines deployment:
+
+1. Make sure `mutatingwebhookconfiguration` exists in your cluster:
+
+    ```
+    export NAMESPACE=<Namespace where KFP is installed>
+    kubectl get mutatingwebhookconfiguration cache-webhook -n ${NAMESPACE}
+    ```
+2. Change `mutatingwebhookconfiguration` rules:
+
+    ```
+    kubectl patch mutatingwebhookconfiguration cache-webhook -n ${NAMESPACE} --type='json' -p='[{"op":"replace", "path": "/webhooks/0/rules/0/operations/0", "value": "DELETE"}]'
+    ```
+
+### Enabling caching
+
+1. Make sure `mutatingwebhookconfiguration` exists in your cluster:
+
+    ```
+    export NAMESPACE=<Namespace where KFP is installed>
+    kubectl get mutatingwebhookconfiguration cache-webhook -n ${NAMESPACE}
+    ```
+2. Change back `mutatingwebhookconfiguration` rules:
+
+    ```
+    kubectl patch mutatingwebhookconfiguration cache-webhook -n ${NAMESPACE} --type='json' -p='[{"op":"replace", "path": "/webhooks/0/rules/0/operations/0", "value": "CREATE"}]'
+    ```
+
+## Managing caching staleness
+
+The cache is enabled by default and if you ever executed same component with the same arguments, any new execution of the component will be skipped and the outputs will be taken from the cache.
+For some scenarios, the cached output data of some components might become too stale for use after some time.
+To control the maximum staleness of the reused cached data, you can set the step's `max_cache_staleness` parameter.
+The `max_cache_staleness` is in [RFC3339 Duration](https://www.ietf.org/rfc/rfc3339.txt) format (so 30 days = "P30D"). 
+By default the `max_cache_staleness` is set to infinity so any old cached data will be reused.
+
+Set `max_cache_staleness` to 30 days for a step:
+
+```
+def some_pipeline():
+      # task is a target step in a pipeline
+      task = some_op()
+      task.execution_options.caching_strategy.max_cache_staleness = "P30D"
+```
+
+Ideally, the component code should be pure and deterministic in the sense that it produces same outputs given same inputs.
```

Comment:
Nit: Isn't a function that is pure deterministic by definition?

Ideally, each component should be pure, in that the component deterministically produces the same outputs for a given set of inputs and does not have additional side effects.